src/LiveryBridge.ts

```diff
@@ -0,0 +1,296 @@
+/* eslint-disable @typescript-eslint/no-explicit-any --- MessageEvent returns an any object. The maps we use also use any, as multiple data types are allowed. */
+
+import { uuid } from './livery-interactive/util/uuid';
+
+export interface LiveryMessage {
+  id?: string;
+  isLivery: true;
+  type: string;
+}
+
+interface HandshakeMessage extends LiveryMessage {
+  id: string;
+  type: 'handshake';
+  version: string;
+}
+
+interface ResolveMessage<ValueType> extends LiveryMessage {
+  id: string;
+  type: 'resolve';
+  value: ValueType;
+}
+
+interface RejectMessage extends LiveryMessage {
+  error: Error;
+  id: string;
+  type: 'reject';
+}
+
+interface CommandMessage extends LiveryMessage {
+  arg?: string;
+  id: string;
+  name: string;
+  type: 'command';
+}
+
+interface EventMessage<ValueType> extends LiveryMessage {
+  id: string;
+  type: 'event';
+  value: ValueType;
+}
+
+interface Deferred {
+  reject: (error: Error) => void;
+  resolve: (value: any) => void;
+}
+
+export class LiveryBridge {
+  protected deferredMap: Map<string, Deferred>;
+
+  protected handshakeId: string;
+
+  protected handshakePromise: Promise<void>;
+
+  protected listenerMap: Map<string, (value: any) => void>;
+
+  protected messageListener?: (message: string) => void;
+
+  protected targetWindow: Window;
+
+  protected targetWindowUrl: string;
+
+  protected version: string;
+
+  constructor(
+    targetWindow: Window,
+    targetWindowUrl: string,
+    version: string,
+    messageListener?: (message: string) => void,
```

Comment:
Don't pass a stringified message, just pass the message object, e.g: `message: LiveryMessage`.

Reply:
In that case I'll also pass a boolean `messageIn`, so the messageListener can differentiate between an incoming and outgoing message. Passing a `title` string to be put before the message might also be an idea. What do you think?